### ISSUE TITLE
Julia interface

### ISSUE THREAD

**Issue by peterwittek** [peterwittek/somoclu#11]
Julia lacks a SOM implementation. Once v0.4 of Julia is released and Cxx.jl matures, it will be worthwhile to make a module.

**Comment by peterwittek** [comment]
The main C++ interface, [Cxx,jl](https://github.com/Keno/Cxx.jl), still requires a development version and the source code of Julia to work, so it is better to wait until this interface matures.

**Comment by psilva07** [comment]
+1 for this!

**Comment by peterwittek** [comment]
According to [this issue](https://github.com/Keno/Cxx.jl/issues/237), there is a hope that when Julia 0.5 is out, Cxx.jl can be installed as a normal package. This release is [due at the end of July](https://github.com/JuliaLang/julia/milestones), so I can look into this in August.

**Comment by peterwittek** [comment]
An extremely rudimentary version of the interface is now available by commit ba2ec3b96f9cfb77e44a4fb8feac19a16621782f. The following code works, albeit it is not particularly useful:

```julia
import Somoclu.train

nSomX, nSomY = 40, 20
nDimensions, nVectors = 3, 20
data = Array{Float32}(nVectors*nDimensions)
codebook = Array{Float32}(nSomX*nSomY*nDimensions)
bmus = Array{Cint}(nVectors*2)
uMatrix = Array{Float32}(nSomX*nSomY)

train(data, codebook, bmus, uMatrix, nDimensions, nVectors, nSomX, nSomY)
```

Build it by
```bash
./configure --without-cuda --without-mpi
make julia
```

I hope to get it in shape by the end of the month.

**Comment by peterwittek** [comment]
The Julia interface was migrated to a [separate repo](https://github.com/peterwittek/Somoclu.jl) to comply with Julia package distribution methods. In principle, it works.
